share/jupyter/nbconvert/templates/base/jupyter_widgets.html.j2

```diff
@@ -1,19 +1,27 @@
-{%- macro jupyter_widgets(widgets_cdn_url, html_manager_semver_range) -%}
+{%- macro jupyter_widgets(widgets_cdn_url, html_manager_semver_range, widgets_url='') -%}
 
 <script>
 (function() {
   function addWidgetsRenderer() {
     var mimeElement = document.querySelector('script[type="application/vnd.jupyter.widget-view+json"]');
     var scriptElement = document.createElement('script');
+    {% if widgets_url %}   
+    var widgetRendererSrc = '{{ widgets_url }}embed-amd.js';
```

Comment:
Since it said the widgets_url was the "full URL for Jupyter widgets", I did the following: `jupyter nbconvert  NB.ipynb --to html --HTMLExporter.widgets_url='https://cdn.jsdelivr.net/npm/@jupyter-widgets/html-manager/dist/embed-amd.js'`

However, that attempted to get `https://cdn.jsdelivr.net/npm/@jupyter-widgets/html-manager/dist/embed-amd.jsembed-amd.js`, which you can see has an extra `embed-amd.js` tacked onto the end.

@bsyouness, do you think it would be more clear to make this url the actual full url, and not tack on the embed-amd.js part?

I see that below, we have another call that changes `embed-amd.js` for `embed.js` for widget state below version 2, but I think that:

1. the widget state was upgraded to version 2 on the ipywidgets 7.0 release, which was Aug 18, 2017, so we probably don't need to automatically switch the fetched file anymore on new changes.
2. If a person is converting an old file, they can manually specify the `.../embed.js` file (since they will need to give an old version of the html manager anyway, I think).

So I'd suggest changing the above line to:

```suggestion
    var widgetRendererSrc = '{{ widgets_url }}';
```

and making a similar change to line 21 below (deleting `embed.js`.

Reply:
I agree with jason on this.

Also widgets_url really concerns the URL for the manager URL, and not the URL where custom widgets are fetched. Maybe the name should be a bit more specific.

Reply:
maybe `widget_renderer_url`...